syntaxes/odoc.json

```diff
@@ -0,0 +1,171 @@
+{
+  "scopeName": "source.ocaml.odoc",
+  "name": "comment.other.odoc",
+  "fileTypes": ["mld"],
+  "patterns": [
+    {
+      "name": "comment.other.odoc",
+      "begin": "",
+      "end": "",
+      "patterns": [{ "include": "#markup" }]
+    },
+    {
+      "comment": "odoc documentation tag",
+      "name": "keyword.doc-tag.odoc",
+      "match": "\\@[[:lower:]]+"
+    }
+  ],
+  "repository": {
+    "markup": {
+      "patterns": [
+        {
+          "comment": "embedded ocaml source",
+          "begin": "(?<!\\\\)\\[",
+          "end": "(?<!\\\\)\\]",
+          "contentName": "source.embedded.odoc",
+          "patterns": [{ "include": "source.ocaml" }]
+        },
+        {
+          "comment": "label",
+          "begin": "\\{([[:digit:]]+)\\:[[:space:]]?[a-z]+",
+          "end": "\\}",
+          "beginCaptures": {
+            "1": { "name": "variable.other.declaration.odoc" }
+          },
```

Comment:
This needs two separate captures and colors: one for the number and one for the label.

You are currently highlighting the number as a variable.